frontend/gateway/gateway.go

```diff
@@ -1177,6 +1179,7 @@ func (w *outputWriter) Write(msg []byte) (int, error) {
 }
 
 func (lbf *llbBridgeForwarder) ExecProcess(srv pb.LLBBridge_ExecProcessServer) error {
+	lbf.srv = srv
```

Comment:
This assignment does not look right. `ExecProcess` can be called multiple times and shouldn't change the state of `llbBridgeForwarder`. The mutex should be defined inline here, or another way would be to just wrap `LLBBridge_ExecProcessServer` so that `Send` method becomes thread-safe.